File: src_fortran/sub/sdc.f
c
c
c
c
c
c
      subroutine SDC(x,y,xlat,xlon,i)     ! 7. Jan. 1988
c
c     SDC needs subroutines SETORG, DIST and REDIST
c
c
      implicit none
      real x,y,xlat,xlon
      integer i
c
      real olat,olon,aa,bb,bc,sint,cost,rotate
      integer icoordsystem
      common/GEO_COORSYSTEM/olat,olon,
     &                      rearth,ellip,rlatc,rad,
     &                      aa,bb,bc,sint,cost,rotate, icoordsystem
      double precision rearth,ellip,rlatc,rad
c
c
cc      call setorg()  ! is called separately !!
c
      if(.not.(i.eq.-1.or.i.eq.1)) stop'SDC>>> specify conversion !!!'
c
      if(i.eq.1)  call redist(x,y,xlat,xlon)
      if(i.eq.-1) call dist(xlat,xlon,x,y)
      write(*,*) 'DEBUG: sdc xlat = ', xlat
      write(*,*) 'DEBUG: sdc xlon = ', xlon
      write(*,*) 'DEBUG: sdc i = ', i
      write(*,*) 'DEBUG: sdc x = ', x
      write(*,*) 'DEBUG: sdc y = ', y
c
      end ! of subr. SDC

File: src/sub/sdc.c
#include<stdio.h>

void dist(double xlat, double xlon, double *xkm, double *ykm);
void redist(double xkm, double ykm, double *xlat, double *xlon);

void sdc(double *x, double *y, double xlat, double xlon, int i) {
    if (!x || !y) {
        return;
    }
    if (i != -1 && i != 1) {
        return;
    }

    if (i == -1) {
        dist(xlat, xlon, x, y);
    } else {
        redist(xlat, xlon, x, y);
    }
    fprintf(stderr, "DEBUG: sdc x0 = %f \n", xlat);
    fprintf(stderr, "DEBUG: sdc y0 = %f \n", xlon);
    fprintf(stderr, "DEBUG: sdc i = %d \n", i);
    fprintf(stderr, "DEBUG: sdc x = %f \n", *x);
    fprintf(stderr, "DEBUG: sdc y = %f \n", *y);
}
